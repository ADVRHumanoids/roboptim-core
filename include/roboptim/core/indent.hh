// Copyright (C) 2009 by Thomas Moulard, AIST, CNRS, INRIA.
//
// This file is part of the roboptim.
//
// roboptim is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// roboptim is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with roboptim.  If not, see <http://www.gnu.org/licenses/>.


/**
 * \brief Declaration of indentation related methods.
 */

#ifndef ROBOPTIM_CORE_INDENT_HH
# define ROBOPTIM_CORE_INDENT_HH
# include <iosfwd>

namespace roboptim
{
  /// Increment the indentation.
  std::ostream& incindent (std::ostream& o);

  /// Decrement the indentation.
  std::ostream& decindent (std::ostream& o);

  /// Reset the indentation.
  std::ostream& resetindent (std::ostream& o);

  /// Print an end of line, then set the indentation.
  std::ostream& iendl (std::ostream& o);

  /// Increment the indentation, print an end of line, and set the indentation.
  std::ostream& incendl (std::ostream& o);

  /// Decrement the indentation, print an end of line, and set the indentation.
  std::ostream& decendl (std::ostream& o);
}

#endif // !MISC_INDENT_HH
