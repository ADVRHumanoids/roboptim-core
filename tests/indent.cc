// Copyright (C) 2014 by Benjamin Chr√©tien, CNRS.
//
// This file is part of the roboptim.
//
// roboptim is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// roboptim is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with roboptim.  If not, see <http://www.gnu.org/licenses/>.

#include "shared-tests/fixture.hh"

#include <iostream>

#include <roboptim/core/io.hh>
#include <roboptim/core/indent.hh>

using namespace roboptim;

BOOST_FIXTURE_TEST_SUITE (core, TestSuiteConfiguration)

BOOST_AUTO_TEST_CASE (indent)
{
  boost::shared_ptr<boost::test_tools::output_test_stream>
    output = retrievePattern ("indent");

  (*output) << "Test 1" << incindent;
  (*output) << iendl << "Dummy 1";
  (*output) << iendl << "Dummy 2";
  (*output) << incindent;
  (*output) << iendl << "Foo";
  (*output) << iendl << "Bar";
  (*output) << decindent;
  (*output) << iendl << "Dummy 3";
  (*output) << resetindent << iendl << "Test 2";
  (*output) << incendl << "Dummy 4";
  (*output) << decendl << "Test 3";

  std::cout << output->str () << std::endl;
  BOOST_CHECK (output->match_pattern ());
}

BOOST_AUTO_TEST_SUITE_END ()
