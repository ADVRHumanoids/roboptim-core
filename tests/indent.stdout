Test 1
  Dummy 1
  Dummy 2
    Foo
    Bar
  Dummy 3
Test 2
  Dummy 4
Test 3
